### PR TITLE
Add supporting sklearn=0.23.2 for nni

### PR DESCRIPTION
**Pull request by linbinskn**
Add supporting sklearn=0.23.2 for nni, test on python3.6 and python3.8 successfully.

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -13,6 +13,6 @@ peewee
 nbsphinx
 schema
 tensorboard
-scikit-learn==0.20
+scikit-learn==0.23.2
```
Comment (chicm-ms) [inline docs/requirements.txt:16]: should change to `scikit-learn>=0.23.2` here and other `requirements.txt`

File: deployment/pypi/setup.py [hunk 2]
```diff
@@ -63,7 +63,7 @@
         'scipy',
         'coverage',
         'colorama',
-        'scikit-learn>=0.20,<0.22',
+        'scikit-learn>=0.23,<=0.24',
```
Comment (chicm-ms) [inline deployment/pypi/setup.py:66]: `<=0.24` can be removed.

Comment (linbinskn) [reply deployment/pypi/setup.py]: Have changed, please reviewe again.

Comment (linbinskn) [reply deployment/pypi/setup.py]: Have changed, please reviewe again.

File: deployment/docker/README_zh_CN.md [hunk 3]
```diff
@@ -9,7 +9,7 @@
     TensorFlow-gpu 1.10.0
     Keras 2.1.6
     PyTorch 0.4.1
-    scikit-learn 0.20.0
+    scikit-learn 0.23.2
```
Comment (QuanluZhang) [inline deployment/docker/README_zh_CN.md:12]: please do not change Chinese doc

File: deployment/pypi/setup.py [hunk 4]
```diff
@@ -63,7 +63,7 @@
         'scipy',
         'coverage',
         'colorama',
-        'scikit-learn>=0.20,<0.22',
+        'scikit-learn>=0.23',
```
Comment (chicm-ms) [inline deployment/pypi/setup.py:66]: why this one is 0.23 and or others are 0.23.2?

Comment (linbinskn) [reply deployment/pypi/setup.py]: Both of they are ok. The 0.23.2 is more clear. Have modified.

**Pull request merged**